dex/router/src/config.rs

```diff
@@ -2,18 +2,38 @@ multiversx_sc::imports!();
 multiversx_sc::derive_imports!();
 
 use crate::{enable_swap_by_user::EnableSwapByUserConfig, factory::PairTokens};
+use pair::read_pair_storage;
 
 #[multiversx_sc::module]
-pub trait ConfigModule {
+pub trait ConfigModule: read_pair_storage::ReadPairStorageModule {
     fn is_active(&self) -> bool {
         self.state().get()
     }
 
     fn check_is_pair_sc(&self, pair_address: &ManagedAddress) {
-        require!(
-            self.address_pair_map().contains_key(pair_address),
-            "Not a pair SC"
-        );
+        let first_token_id = self.get_first_token_id_mapper(pair_address.clone()).get();
+        let second_token_id = self.get_second_token_id_mapper(pair_address.clone()).get();
+
+        let pair_tokens = PairTokens {
+            first_token_id: first_token_id.clone(),
+            second_token_id: second_token_id.clone(),
+        };
+
+        let mut pair_map_address_opt = self.pair_map().get(&pair_tokens);
+        if pair_map_address_opt.is_none() {
```

Comment:
you should not check for the reverse order of tokens. If the pair address is not found with the correct order provided by the pair storage, it will not work for sure

Reply:
Removed.

Reply:
Added back after feedback.